hail/Makefile

```diff
@@ -324,7 +324,7 @@ install: $(WHEEL)
 
 .PHONY: install-on-cluster
 install-on-cluster: $(WHEEL)
```

Comment:
this target can run without generated pinned requirements files.

Reply:
Not sure I understand entirely. Do you mean that `install-on-cluster` should use unpinned requirements or is this a concern about the target / dependencies of the target?

Reply:
This is similar to my concerns above, there's an implicit dependency between python/requirements.txt and python/pinned-requirements.txt. At the very least we should error if the pinned-requirements.txt is older than the requirements.txt